src/Data/Quaternion.purs

```diff
@@ -128,9 +128,14 @@ instance ringQuaternion :: Ring a => Ring (Quaternion a) where
   sub (Quaternion a1 b1 c1 d1) (Quaternion a2 b2 c2 d2) =
     Quaternion (a1 - a2) (b1 - b2) (c1 - c2) (d1 - d2)
 
+
+instance functorQuaternion :: Functor Quaternion where
+  map f (Quaternion a b c d) = Quaternion (f a) (f b) (f c) (f d)
```

Comment:
Up to you of course, but there's nothing wrong with small libraries; this one only exports a single function! https://pursuit.purescript.org/packages/purescript-unsafe-coerce/